lib/matplotlib/backend_bases.py

```diff
@@ -3060,14 +3060,62 @@ def press_pan(self, event):
         if (event.button not in [MouseButton.LEFT, MouseButton.RIGHT]
                 or event.x is None or event.y is None):
             return
-        axes = [a for a in self.canvas.figure.get_axes()
-                if a.in_axes(event) and a.get_navigate() and a.can_pan()]
-        if not axes:
+
+        # Get all axes and group them by their navigation-capture behaviour.
+        axes_capture = {True: {}, False: {}}
```

Comment:
I'm still not certain why this is necessary. Why can't you sort by zorder and stop at the first Axes that captures?

Reply:
hey, so I tried to explain this before very briefly here: https://github.com/matplotlib/matplotlib/pull/22347#discussion_r795041340

I'll try to elaborate a bit more on this... the main reason why I implemented it is the following:
- assume you have an axis (say `ax1`) with a visible patch (e.g. a capturing axis)
- and a (maybe but not necessarily overlapping) axis (`ax2`) that also has a twin-axis (`ax2_twin`)

now, if you pan/zoom `ax1`, the following happens:
- if `ax2` is **at the same or higher zorder**, `ax2_twin` would always be triggered because `ax2_twin` has **no visible patch** (e.g. it's non-capturing just as any other twin-axis).
- to avoid this, the implemented logic is the following:
  - first trigger the topmost (capturing) axes (e.g. in `axes_capture[True][zorder]`) and all twin-axes of it  
    (those ones we 100% always want to trigger)
  - then, check for non-capturing axes (e.g. no visible patch) at the same or higher zorder  
    (`axes_capture[False][>=zorder]`)
    - if an axis is actually just a twin-axis of an axis with a visible patch (e.g. its parent is in `axes_capture[True]`)  **don't trigger it** (note that we would have triggered it already as sibling of it's parent if relevant! )
    - otherwise trigger it (if it has not already been triggered)

So far this was the most easy way I could think of to ensure triggering only axes we really want to trigger...

Reply:
I have some concerns about the "same or higher zorder" part of this.  In all other cases we sort on zoreder using a stable sort so we always fully break the degeneracy and based on the order the Artists were added.  That is to say, even though two Axes (or any artist) may have identical zorder's there is a cannonical ordering and one is ahead of the other. 

I think the case that is prolemaatic here is `ax2_twin > ax1 > ax2` (that is ax1 is "between" the two layers of of the twin axes and you want to trigger all three) vs `ax2_twin > ax2 > ax1` and `ax1 > ax2_twin > ax2` where you only want to trigger ax1 or the ax2 set?

I am also very late to the party here and have not fully read up so I could me missunderstanding something....

Reply:
@tacaswell 😄 I tried to capture as many possibilities as possible in the test-plot... 
concerning your question, twin-axes are treated explicitly so all mentioned cases are already properly handled!